Add local segregation measures

renanxcortes opened the pull request:
This is a kickstart for local segregation measures :)
I still need to work on other stuff (that is why this is still a WIP) like:

- ReadMe
- docstrings
- tests
- Sphinx/docs build
- perhaps add maps to example notebook
- perhaps return an attached column to the input dataset rather than a numpy array

I wanted to highlight that all the values are matching the OasisR implementations. In their case, they only work with the Simpson Interaction Index, whereas we have the interaction and the concentration. 

The Local Centralization Method follows strictly Folch, David C., and Sergio J. Rey. "The centralization index: A measure of local spatial segregation." Papers in Regional Science 95.3 (2016): 555-576. Currently, I only added the k nearest neighborhood for local context. In addition, the approach of this paper is basically defining a local context and apply the Relative Centralization index in this local context. I think that this approach can be extended basically to any kind of segregation measure, right? In this context we can get "for free":

- local Dissimilarity
- local Gini
- local Centralization/Concentration
- etc, etc.

So, I think I could add these measures in the future, by basically wrapping the global measures to this adaptation. I highlight also that I thought about using `_build_local_environment` of https://github.com/pysal/segregation/pull/75/commits/01a3fc521340157de603d757836e9ab4ba808479, however, the approach here is substantially different since we need to "filter" the local environment rather than getting the spatial lag (more details in Section 3 of the mentioned paper)

I'll keep working on this in the following days and will let you know when I'm ok for the review and, eventual, merge.

Cheers,
Renan

renanxcortes commented:
@knaaptime and @sjsrey I think this PR is good to go. Could any of you take a look at it? In a nutshell, it adds a new set of local segregation measures. The description is the first comment of this PR. The more recent commits is just to update everything that depends on it (readthedocs, tests, docstrings, etc.)

knaaptime commented:
- what about adding a quick plot in the notebook for each set of local measures?
- i added this line to each module `np.seterr(divide='ignore', invalid='ignore')` to avoid the numpy warnings, which aren't really useful because we know there will be some divisions by zero. Could you add the same in local_indexes.py?

renanxcortes commented:
> * what about adding a quick plot in the notebook for each set of local measures?
> * i added this line to each module `np.seterr(divide='ignore', invalid='ignore')` to avoid the numpy warnings, which aren't really useful because we know there will be some divisions by zero. Could you add the same in local_indexes.py?

Sure, will do!

knaaptime commented on segregation/local/local_indexes.py:
```diff
@@ -12,6 +12,10 @@
+
 
 __all__ = [
     'Multi_Location_Quocient',
```
this should be "Quotient"

renanxcortes replied:
Ow, sure, will fix this!

renanxcortes replied:
> this should be "Quotient"

I think with the last commit, this is addressed. BTW, in Brazilian Portuguese, we use "quociente", that's why I got confused! hahaha

renanxcortes commented:
hm... the TreaviCI broke somehow and I don`t know really why.. it seems like it broke during the creation of the environment, but this wasn't happening before... maybe after we merge https://github.com/pysal/segregation/pull/78 that deals with the structure would this be solved.

Currently the error is: 

```
Worker information
hostname: 20d97444-11f2-41ce-992a-1d692f7d3268@1.production-2-worker-com-gce-ptcp
version: v6.2.0 https://github.com/travis-ci/worker/tree/5e5476e01646095f48eec13196fdb3faf8f5cbf7
instance: travis-job-66ada128-44bc-4b2a-8a29-1d091b10085c travis-ci-sardonyx-xenial-1553530528-f909ac5 (via amqp)
startup: 6.565640392s
system_info
Build system information
Build language: python
Build group: stable
Build dist: xenial
Build id: 116891444
Job id: 210967838
Runtime kernel version: 4.15.0-1028-gcp
travis-build version: 49f4a9604
Build image provisioning date and time
Mon Mar 25 16:43:24 UTC 2019
Operating System Details
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.6 LTS
Release:	16.04
Codename:	xenial
Systemd Version
systemd 229
Cookbooks Version
42e42e4 https://github.com/travis-ci/travis-cookbooks/tree/42e42e4
git version
git version 2.21.0
bash version
GNU bash, version 4.3.48(1)-release (x86_64-pc-linux-gnu)
gcc version
gcc (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609
docker version
Client:
 Version:           18.06.0-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        0ffa825
 Built:             Wed Jul 18 19:11:02 2018
 OS/Arch:           linux/amd64
 Experimental:      false
Server:
 Engine:
  Version:          18.06.0-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       0ffa825
  Built:            Wed Jul 18 19:09:05 2018
  OS/Arch:          linux/amd64
  Experimental:     false
clang version
clang version 7.0.0 (tags/RELEASE_700/final)
jq version
jq-1.5
bats version
Bats 0.4.0
shellcheck version
0.6.0
shfmt version
v2.6.3
ccache version
3.2.4
cmake version
cmake version 3.12.4
heroku version
heroku/7.22.7 linux-x64 node-v11.10.1
imagemagick version
Version: ImageMagick 6.8.9-9 Q16 x86_64 2018-09-28 http://www.imagemagick.org
md5deep version
4.4
mercurial version
version 4.8
mysql version
mysql  Ver 14.14 Distrib 5.7.25, for Linux (x86_64) using  EditLine wrapper
openssl version
OpenSSL 1.0.2g  1 Mar 2016
packer version
1.3.3
postgresql client version
psql (PostgreSQL) 10.7 (Ubuntu 10.7-1.pgdg16.04+1)
ragel version
Ragel State Machine Compiler version 6.8 Feb 2013
sudo version
1.8.16
gzip version
gzip 1.6
zip version
Zip 3.0
vim version
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Nov 24 2016 16:44:48)
iptables version
iptables v1.6.0
curl version
curl 7.47.0 (x86_64-pc-linux-gnu) libcurl/7.47.0 GnuTLS/3.4.10 zlib/1.2.8 libidn/1.32 librtmp/2.3
wget version
GNU Wget 1.17.1 built on linux-gnu.
rsync version
rsync  version 3.1.1  protocol version 31
gimme version
v1.5.3
nvm version
0.34.0
perlbrew version
/home/travis/perl5/perlbrew/bin/perlbrew  - App::perlbrew/0.86
phpenv version
rbenv 1.1.2
rvm version
rvm 1.29.7 (latest) by Michal Papis, Piotr Kuczynski, Wayne E. Seguin [https://rvm.io]
default ruby version
ruby 2.5.3p105 (2018-10-18 revision 65156) [x86_64-linux]
CouchDB version
couchdb 1.6.1
ElasticSearch version
5.5.0
Installed Firefox version
firefox 63.0.1
MongoDB version
MongoDB 4.0.7
PhantomJS version
2.1.1
Pre-installed PostgreSQL versions
9.4.21
9.5.16
9.6.12
Redis version
redis-server 5.0.4
Pre-installed Go versions
1.11.1
mvn version
Apache Maven 3.6.0 (97c98ec64a1fdfee7767ce5ffb20918da4f719f3; 2018-10-24T18:41:47Z)
gradle version
Gradle 4.10.2!
lein version
Leiningen 2.9.1 on Java 11.0.2 OpenJDK 64-Bit Server VM
Pre-installed Node.js versions
v10.15.3
v11.0.0
v4.9.1
v6.17.0
v8.12.0
v8.15.1
v8.9
phpenv versions
  system
  5.6
  5.6.40
  7.1
  7.1.27
  7.2
* 7.2.15 (set by /home/travis/.phpenv/version)
  hhvm
  hhvm-stable
composer --version
Composer version 1.8.4 2019-02-11 10:52:10
Pre-installed Ruby versions
ruby-2.3.8
ruby-2.4.5
ruby-2.5.3
docker_mtu
resolvconf
git.checkout
2.44s$ git clone --depth=50 https://github.com/pysal/segregation.git pysal/segregation
Cloning into 'pysal/segregation'...
remote: Enumerating objects: 1404, done.
remote: Counting objects: 100% (1404/1404), done.
remote: Compressing objects: 100% (526/526), done.
remote: Total 1404 (delta 881), reused 1346 (delta 836), pack-reused 0
Receiving objects: 100% (1404/1404), 24.26 MiB | 28.20 MiB/s, done.
Resolving deltas: 100% (881/881), done.
$ cd pysal/segregation
0.76s$ git fetch origin +refs/pull/77/merge:
remote: Enumerating objects: 146, done.
remote: Counting objects: 100% (137/137), done.
remote: Compressing objects: 100% (48/48), done.
remote: Total 120 (delta 79), reused 108 (delta 70), pack-reused 0
Receiving objects: 100% (120/120), 761.81 KiB | 7.05 MiB/s, done.
Resolving deltas: 100% (79/79), completed with 16 local objects.
From https://github.com/pysal/segregation
 * branch            refs/pull/77/merge -> FETCH_HEAD
$ git checkout -qf FETCH_HEAD
Setting environment variables from .travis.yml
$ export PYSAL_PYPI=false
0.01s$ source ~/virtualenv/python3.7/bin/activate
$ python --version
Python 3.7.1
$ pip --version
pip 19.0.3 from /home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/pip (python 3.7)
before_install.1
0.54s$ wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
--2019-06-25 22:32:59--  https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
Resolving repo.continuum.io (repo.continuum.io)... 104.18.200.79, 104.18.201.79, 2606:4700::6812:c84f, ...
Connecting to repo.continuum.io (repo.continuum.io)|104.18.200.79|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 70348401 (67M) [application/x-sh]
Saving to: ‘miniconda.sh’
miniconda.sh        100%[===================>]  67.09M   156MB/s    in 0.4s    
2019-06-25 22:32:59 (156 MB/s) - ‘miniconda.sh’ saved [70348401/70348401]
before_install.2
0.00s$ chmod +x miniconda.sh
before_install.3
12.62s$ ./miniconda.sh -b -p ./miniconda
PREFIX=/home/travis/build/pysal/segregation/miniconda
installing: python-3.7.3-h0371630_0 ...
Python 3.7.3
installing: ca-certificates-2019.1.23-0 ...
installing: libgcc-ng-8.2.0-hdf63c60_1 ...
installing: libstdcxx-ng-8.2.0-hdf63c60_1 ...
installing: libffi-3.2.1-hd88cf55_4 ...
installing: ncurses-6.1-he6710b0_1 ...
installing: openssl-1.1.1b-h7b6447c_1 ...
installing: xz-5.2.4-h14c3975_4 ...
installing: yaml-0.1.7-had09818_2 ...
installing: zlib-1.2.11-h7b6447c_3 ...
installing: libedit-3.1.20181209-hc058e9b_0 ...
installing: readline-7.0-h7b6447c_5 ...
installing: tk-8.6.8-hbc83047_0 ...
installing: sqlite-3.27.2-h7b6447c_0 ...
installing: asn1crypto-0.24.0-py37_0 ...
installing: certifi-2019.3.9-py37_0 ...
installing: chardet-3.0.4-py37_1 ...
installing: idna-2.8-py37_0 ...
installing: pycosat-0.6.3-py37h14c3975_0 ...
installing: pycparser-2.19-py37_0 ...
installing: pysocks-1.6.8-py37_0 ...
installing: ruamel_yaml-0.15.46-py37h14c3975_0 ...
installing: six-1.12.0-py37_0 ...
installing: cffi-1.12.2-py37h2e261b9_1 ...
installing: setuptools-41.0.0-py37_0 ...
installing: cryptography-2.6.1-py37h1ba5d50_0 ...
installing: wheel-0.33.1-py37_0 ...
installing: pip-19.0.3-py37_0 ...
installing: pyopenssl-19.0.0-py37_0 ...
installing: urllib3-1.24.1-py37_0 ...
installing: requests-2.21.0-py37_0 ...
installing: conda-4.6.14-py37_0 ...
installation finished.
before_install.4
0.00s$ export PATH=`pwd`/miniconda/bin:$PATH
before_install.5
19.43s$ conda update --yes conda
Collecting package metadata: done
Solving environment: done
## Package Plan ##
  environment location: /home/travis/build/pysal/segregation/miniconda
  added / updated specs:
    - conda
The following packages will be downloaded:
    package                    |            build
    ---------------------------|-----------------
    bzip2-1.0.6                |       h14c3975_5         414 KB
    ca-certificates-2019.5.15  |                0         133 KB
    certifi-2019.6.16          |           py37_0         154 KB
    conda-4.7.5                |           py37_0         3.0 MB
    conda-package-handling-1.3.10|           py37_0         259 KB
    icu-58.2                   |       h9c2bf20_1        22.5 MB
    libarchive-3.3.3           |       h5d8350f_5         1.5 MB
    libxml2-2.9.9              |       he19cac6_0         2.0 MB
    lz4-c-1.8.1.2              |       h14c3975_0         158 KB
    lzo-2.10                   |       h49e0be7_2         313 KB
    openssl-1.1.1c             |       h7b6447c_1         3.8 MB
    python-libarchive-c-2.8    |           py37_6          20 KB
    tqdm-4.32.1                |             py_0          48 KB
    zstd-1.3.7                 |       h0b5b093_0         887 KB
    ------------------------------------------------------------
                                           Total:        35.1 MB
The following NEW packages will be INSTALLED:
  bzip2              pkgs/main/linux-64::bzip2-1.0.6-h14c3975_5
  conda-package-han~ pkgs/main/linux-64::conda-package-handling-1.3.10-py37_0
  icu                pkgs/main/linux-64::icu-58.2-h9c2bf20_1
  libarchive         pkgs/main/linux-64::libarchive-3.3.3-h5d8350f_5
  libxml2            pkgs/main/linux-64::libxml2-2.9.9-he19cac6_0
  lz4-c              pkgs/main/linux-64::lz4-c-1.8.1.2-h14c3975_0
  lzo                pkgs/main/linux-64::lzo-2.10-h49e0be7_2
  python-libarchive~ pkgs/main/linux-64::python-libarchive-c-2.8-py37_6
  tqdm               pkgs/main/noarch::tqdm-4.32.1-py_0
  zstd               pkgs/main/linux-64::zstd-1.3.7-h0b5b093_0
The following packages will be UPDATED:
  ca-certificates                               2019.1.23-0 --> 2019.5.15-0
  certifi                                   2019.3.9-py37_0 --> 2019.6.16-py37_0
  conda                                       4.6.14-py37_0 --> 4.7.5-py37_0
  openssl                                 1.1.1b-h7b6447c_1 --> 1.1.1c-h7b6447c_1
Downloading and Extracting Packages
lzo-2.10             | 313 KB    | ##################################### | 100% 
lz4-c-1.8.1.2        | 158 KB    | ##################################### | 100% 
python-libarchive-c- | 20 KB     | ##################################### | 100% 
openssl-1.1.1c       | 3.8 MB    | ##################################### | 100% 
libxml2-2.9.9        | 2.0 MB    | ##################################### | 100% 
conda-package-handli | 259 KB    | ##################################### | 100% 
ca-certificates-2019 | 133 KB    | ##################################### | 100% 
bzip2-1.0.6          | 414 KB    | ##################################### | 100% 
libarchive-3.3.3     | 1.5 MB    | ##################################### | 100% 
certifi-2019.6.16    | 154 KB    | ##################################### | 100% 
zstd-1.3.7           | 887 KB    | ##################################### | 100% 
conda-4.7.5          | 3.0 MB    | ##################################### | 100% 
icu-58.2             | 22.5 MB   | ##################################### | 100% 
tqdm-4.32.1          | 48 KB     | ##################################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
before_install.6
0.13s$ conda config --add channels conda-forge
before_install.7
0.14s$ conda config --add channels udst
before_install.8
0.13s$ conda config --set channel_priority strict
before_install.9
17.67s$ conda create -y -q -n test-env python=$TRAVIS_PYTHON_VERSION
Collecting package metadata (current_repodata.json): ...working... done
Solving environment: ...working... done
## Package Plan ##
  environment location: /home/travis/build/pysal/segregation/miniconda/envs/test-env
  added / updated specs:
    - python=3.7
The following packages will be downloaded:
    package                    |            build
    ---------------------------|-----------------
    bzip2-1.0.6                |    h14c3975_1002         415 KB  conda-forge
    ca-certificates-2019.6.16  |       hecc5488_0         145 KB  conda-forge
    certifi-2019.6.16          |           py37_0         148 KB  conda-forge
    libffi-3.2.1               |    he1b5a44_1006          46 KB  conda-forge
    libgcc-ng-9.1.0            |       hdf63c60_0         5.1 MB
    libstdcxx-ng-9.1.0         |       hdf63c60_0         3.1 MB
    ncurses-6.1                |    hf484d3e_1002         1.3 MB  conda-forge
    openssl-1.1.1b             |       h14c3975_1         4.0 MB  conda-forge
    pip-19.1.1                 |           py37_0         1.8 MB  conda-forge
    python-3.7.3               |       h5b0a415_0        35.7 MB  conda-forge
    readline-7.0               |    hf8c457e_1001         391 KB  conda-forge
    setuptools-41.0.1          |           py37_0         616 KB  conda-forge
    sqlite-3.28.0              |       h8b20d00_0         1.9 MB  conda-forge
    tk-8.6.9                   |    hed695b0_1002         3.2 MB  conda-forge
    wheel-0.33.4               |           py37_0          34 KB  conda-forge
    xz-5.2.4                   |    h14c3975_1001         366 KB  conda-forge
    zlib-1.2.11                |    h14c3975_1004         101 KB  conda-forge
    ------------------------------------------------------------
                                           Total:        58.3 MB
The following NEW packages will be INSTALLED:
  bzip2              conda-forge/linux-64::bzip2-1.0.6-h14c3975_1002
  ca-certificates    conda-forge/linux-64::ca-certificates-2019.6.16-hecc5488_0
  certifi            conda-forge/linux-64::certifi-2019.6.16-py37_0
  libffi             conda-forge/linux-64::libffi-3.2.1-he1b5a44_1006
  libgcc-ng          pkgs/main/linux-64::libgcc-ng-9.1.0-hdf63c60_0
  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-9.1.0-hdf63c60_0
  ncurses            conda-forge/linux-64::ncurses-6.1-hf484d3e_1002
  openssl            conda-forge/linux-64::openssl-1.1.1b-h14c3975_1
  pip                conda-forge/linux-64::pip-19.1.1-py37_0
  python             conda-forge/linux-64::python-3.7.3-h5b0a415_0
  readline           conda-forge/linux-64::readline-7.0-hf8c457e_1001
  setuptools         conda-forge/linux-64::setuptools-41.0.1-py37_0
  sqlite             conda-forge/linux-64::sqlite-3.28.0-h8b20d00_0
  tk                 conda-forge/linux-64::tk-8.6.9-hed695b0_1002
  wheel              conda-forge/linux-64::wheel-0.33.4-py37_0
  xz                 conda-forge/linux-64::xz-5.2.4-h14c3975_1001
  zlib               conda-forge/linux-64::zlib-1.2.11-h14c3975_1004
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
before_install.10
0.11s$ source activate test-env
install.1
2.76s$ conda install --yes pip nose
Collecting package metadata (current_repodata.json): done
Solving environment: done
## Package Plan ##
  environment location: /home/travis/build/pysal/segregation/miniconda/envs/test-env
  added / updated specs:
    - nose
    - pip
The following packages will be downloaded:
    package                    |            build
    ---------------------------|-----------------
    nose-1.3.7                 |        py37_1002         214 KB  conda-forge
    ------------------------------------------------------------
                                           Total:         214 KB
The following NEW packages will be INSTALLED:
  nose               conda-forge/linux-64::nose-1.3.7-py37_1002
Downloading and Extracting Packages
nose-1.3.7           | 214 KB    | ##################################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
install.2
89.13s$ conda install --yes -c conda-forge geopandas
Collecting package metadata (current_repodata.json): done
Solving environment: done
## Package Plan ##
  environment location: /home/travis/build/pysal/segregation/miniconda/envs/test-env
  added / updated specs:
    - geopandas
The following packages will be downloaded:
    package                    |            build
    ---------------------------|-----------------
    attrs-19.1.0               |             py_0          32 KB  conda-forge
    boost-cpp-1.70.0           |       ha2d47e9_0        21.0 MB  conda-forge
    cairo-1.16.0               |    h18b612c_1001         1.5 MB  conda-forge
    cfitsio-3.430              |    h0bd2449_1002         3.1 MB  conda-forge
    click-7.0                  |             py_0          61 KB  conda-forge
    click-plugins-1.1.1        |             py_0           9 KB  conda-forge
    cligj-0.5.0                |             py_0           8 KB  conda-forge
    curl-7.64.1                |       hf8cf82a_0         128 KB  conda-forge
    cycler-0.10.0              |             py_1           8 KB  conda-forge
    dbus-1.13.6                |       he372182_0         602 KB  conda-forge
    descartes-1.1.0            |             py_3           6 KB  conda-forge
    expat-2.2.5                |    he1b5a44_1003         191 KB  conda-forge
    fiona-1.8.6                |   py37hf242f0b_3         1.1 MB  conda-forge
    fontconfig-2.13.1          |    he4413a7_1000         327 KB  conda-forge
    freetype-2.10.0            |       he983fc9_0         885 KB  conda-forge
    freexl-1.0.5               |    h14c3975_1002          43 KB  conda-forge
    gdal-2.4.1                 |   py37h5f563d9_4         1.4 MB  conda-forge
    geopandas-0.5.0            |             py_2         881 KB  conda-forge
    geos-3.7.1                 |    hf484d3e_1000         1.6 MB  conda-forge
    geotiff-1.5.1              |       h560c3f3_2         279 KB  conda-forge
    gettext-0.19.8.1           |    hc5be6a0_1002         3.6 MB  conda-forge
    giflib-5.1.7               |       h516909a_1         455 KB  conda-forge
    glib-2.58.3                |    h6f030ca_1001         3.3 MB  conda-forge
    gst-plugins-base-1.14.5    |       h0935bb2_0         6.8 MB  conda-forge
    gstreamer-1.14.5           |       h36ae1b5_0         4.5 MB  conda-forge
    hdf4-4.2.13                |    h9a582f1_1002         952 KB  conda-forge
    hdf5-1.10.5                |nompi_h3c11f04_1100         5.2 MB  conda-forge
    icu-58.2                   |    hf484d3e_1000        22.6 MB  conda-forge
    jpeg-9c                    |    h14c3975_1001         251 KB  conda-forge
    json-c-0.13.1              |    h14c3975_1001          71 KB  conda-forge
    kealib-1.4.10              |    h58c409b_1005         173 KB  conda-forge
    kiwisolver-1.1.0           |   py37hc9558a2_0          86 KB  conda-forge
    krb5-1.16.3                |    h05b26f9_1001         1.4 MB  conda-forge
    libblas-3.8.0              |       7_openblas           6 KB  conda-forge
    libcblas-3.8.0             |       7_openblas           6 KB  conda-forge
    libcurl-7.64.1             |       hda55be3_0         584 KB  conda-forge
    libdap4-3.19.1             |    hd48c02d_1000         1.6 MB  conda-forge
    libedit-3.1.20170329       |    hf8c457e_1001         172 KB  conda-forge
    libgdal-2.4.1              |       h654d158_4        18.7 MB  conda-forge
    libgfortran-ng-7.3.0       |       hdf63c60_0        1006 KB
    libiconv-1.15              |    h516909a_1005         2.0 MB  conda-forge
    libkml-1.3.0               |    h4fcabce_1010         643 KB  conda-forge
    liblapack-3.8.0            |       7_openblas           6 KB  conda-forge
    libnetcdf-4.6.2            |    h056eaf5_1002         1.3 MB  conda-forge
    libpng-1.6.37              |       hed695b0_0         343 KB  conda-forge
    libpq-11.3                 |       h4e4e079_0         2.5 MB  conda-forge
    libspatialindex-1.9.0      |       he1b5a44_1         3.1 MB  conda-forge
    libspatialite-4.3.0a       |    h4c5cd43_1028         3.1 MB  conda-forge
    libssh2-1.8.2              |       h22169c7_2         257 KB  conda-forge
    libtiff-4.0.10             |    h57b8799_1003         587 KB  conda-forge
    libuuid-2.32.1             |    h14c3975_1000          26 KB  conda-forge
    libxcb-1.13                |    h14c3975_1002         396 KB  conda-forge
    libxml2-2.9.9              |       h13577e0_0         2.0 MB  conda-forge
    lz4-c-1.8.3                |    he1b5a44_1001         187 KB  conda-forge
    mapclassify-2.0.1          |             py_0          24 KB  conda-forge
    matplotlib-3.1.0           |           py37_1           6 KB  conda-forge
    matplotlib-base-3.1.0      |   py37hfd891ef_1         6.6 MB  conda-forge
    munch-2.3.2                |             py_0           8 KB  conda-forge
    numpy-1.16.4               |   py37h95a1406_0         4.3 MB  conda-forge
    openblas-0.3.5             |    h9ac9557_1001        15.8 MB  conda-forge
    openjpeg-2.3.1             |       h58a6597_0         470 KB  conda-forge
    pandas-0.24.2              |   py37hb3f55d8_0        11.1 MB  conda-forge
    pcre-8.41                  |    hf484d3e_1003         249 KB  conda-forge
    pixman-0.38.0              |    h516909a_1003         594 KB  conda-forge
    poppler-0.67.0             |       h2ad7f00_6         8.8 MB  conda-forge
    poppler-data-0.4.9         |                1         3.4 MB  conda-forge
    postgresql-11.3            |       hbeee2d4_0         4.8 MB  conda-forge
    proj4-6.1.0                |       he751ad9_2         9.4 MB  conda-forge
    pthread-stubs-0.4          |    h14c3975_1001           5 KB  conda-forge
    pyparsing-2.4.0            |             py_0          55 KB  conda-forge
    pyproj-2.2.1               |   py37hc44880f_0         252 KB  conda-forge
    pyqt-5.9.2                 |   py37hcca6a23_0         5.7 MB  conda-forge
    python-dateutil-2.8.0      |             py_0         219 KB  conda-forge
    pytz-2019.1                |             py_0         227 KB  conda-forge
    qt-5.9.7                   |       h52cfd70_2        85.9 MB  conda-forge
    rtree-0.8.3                |py37h666c49c_1002          48 KB  conda-forge
    scipy-1.3.0                |   py37h921218d_0        18.8 MB  conda-forge
    shapely-1.6.4              |py37h06cd6f9_1005         329 KB  conda-forge
    sip-4.19.8                 |py37hf484d3e_1000         290 KB  conda-forge
    six-1.12.0                 |        py37_1000          22 KB  conda-forge
    tornado-6.0.3              |   py37h516909a_0         637 KB  conda-forge
    tzcode-2019a               |    h516909a_1002         429 KB  conda-forge
    xerces-c-3.2.2             |    hac72e42_1001         3.3 MB  conda-forge
    xorg-kbproto-1.0.7         |    h14c3975_1002          26 KB  conda-forge
    xorg-libice-1.0.9          |    h516909a_1004          58 KB  conda-forge
    xorg-libsm-1.2.3           |    h84519dc_1000          25 KB  conda-forge
    xorg-libx11-1.6.7          |    h14c3975_1000         940 KB  conda-forge
    xorg-libxau-1.0.9          |       h14c3975_0          13 KB  conda-forge
    xorg-libxdmcp-1.1.3        |       h516909a_0          18 KB  conda-forge
    xorg-libxext-1.3.4         |       h516909a_0          51 KB  conda-forge
    xorg-libxrender-0.9.10     |    h516909a_1002          31 KB  conda-forge
    xorg-renderproto-0.11.1    |    h14c3975_1002           8 KB  conda-forge
    xorg-xextproto-7.3.0       |    h14c3975_1002          27 KB  conda-forge
    xorg-xproto-7.0.31         |    h14c3975_1007          72 KB  conda-forge
    zstd-1.4.0                 |       h3b9ef0a_0         928 KB  conda-forge
    ------------------------------------------------------------
                                           Total:       304.6 MB
The following NEW packages will be INSTALLED:
  attrs              conda-forge/noarch::attrs-19.1.0-py_0
  boost-cpp          conda-forge/linux-64::boost-cpp-1.70.0-ha2d47e9_0
  cairo              conda-forge/linux-64::cairo-1.16.0-h18b612c_1001
  cfitsio            conda-forge/linux-64::cfitsio-3.430-h0bd2449_1002
  click              conda-forge/noarch::click-7.0-py_0
  click-plugins      conda-forge/noarch::click-plugins-1.1.1-py_0
  cligj              conda-forge/noarch::cligj-0.5.0-py_0
  curl               conda-forge/linux-64::curl-7.64.1-hf8cf82a_0
  cycler             conda-forge/noarch::cycler-0.10.0-py_1
  dbus               conda-forge/linux-64::dbus-1.13.6-he372182_0
  descartes          conda-forge/noarch::descartes-1.1.0-py_3
  expat              conda-forge/linux-64::expat-2.2.5-he1b5a44_1003
  fiona              conda-forge/linux-64::fiona-1.8.6-py37hf242f0b_3
  fontconfig         conda-forge/linux-64::fontconfig-2.13.1-he4413a7_1000
  freetype           conda-forge/linux-64::freetype-2.10.0-he983fc9_0
  freexl             conda-forge/linux-64::freexl-1.0.5-h14c3975_1002
  gdal               conda-forge/linux-64::gdal-2.4.1-py37h5f563d9_4
  geopandas          conda-forge/noarch::geopandas-0.5.0-py_2
  geos               conda-forge/linux-64::geos-3.7.1-hf484d3e_1000
  geotiff            conda-forge/linux-64::geotiff-1.5.1-h560c3f3_2
  gettext            conda-forge/linux-64::gettext-0.19.8.1-hc5be6a0_1002
  giflib             conda-forge/linux-64::giflib-5.1.7-h516909a_1
  glib               conda-forge/linux-64::glib-2.58.3-h6f030ca_1001
  gst-plugins-base   conda-forge/linux-64::gst-plugins-base-1.14.5-h0935bb2_0
  gstreamer          conda-forge/linux-64::gstreamer-1.14.5-h36ae1b5_0
  hdf4               conda-forge/linux-64::hdf4-4.2.13-h9a582f1_1002
  hdf5               conda-forge/linux-64::hdf5-1.10.5-nompi_h3c11f04_1100
  icu                conda-forge/linux-64::icu-58.2-hf484d3e_1000
  jpeg               conda-forge/linux-64::jpeg-9c-h14c3975_1001
  json-c             conda-forge/linux-64::json-c-0.13.1-h14c3975_1001
  kealib             conda-forge/linux-64::kealib-1.4.10-h58c409b_1005
  kiwisolver         conda-forge/linux-64::kiwisolver-1.1.0-py37hc9558a2_0
  krb5               conda-forge/linux-64::krb5-1.16.3-h05b26f9_1001
  libblas            conda-forge/linux-64::libblas-3.8.0-7_openblas
  libcblas           conda-forge/linux-64::libcblas-3.8.0-7_openblas
  libcurl            conda-forge/linux-64::libcurl-7.64.1-hda55be3_0
  libdap4            conda-forge/linux-64::libdap4-3.19.1-hd48c02d_1000
  libedit            conda-forge/linux-64::libedit-3.1.20170329-hf8c457e_1001
  libgdal            conda-forge/linux-64::libgdal-2.4.1-h654d158_4
  libgfortran-ng     pkgs/main/linux-64::libgfortran-ng-7.3.0-hdf63c60_0
  libiconv           conda-forge/linux-64::libiconv-1.15-h516909a_1005
  libkml             conda-forge/linux-64::libkml-1.3.0-h4fcabce_1010
  liblapack          conda-forge/linux-64::liblapack-3.8.0-7_openblas
  libnetcdf          conda-forge/linux-64::libnetcdf-4.6.2-h056eaf5_1002
  libpng             conda-forge/linux-64::libpng-1.6.37-hed695b0_0
  libpq              conda-forge/linux-64::libpq-11.3-h4e4e079_0
  libspatialindex    conda-forge/linux-64::libspatialindex-1.9.0-he1b5a44_1
  libspatialite      conda-forge/linux-64::libspatialite-4.3.0a-h4c5cd43_1028
  libssh2            conda-forge/linux-64::libssh2-1.8.2-h22169c7_2
  libtiff            conda-forge/linux-64::libtiff-4.0.10-h57b8799_1003
  libuuid            conda-forge/linux-64::libuuid-2.32.1-h14c3975_1000
  libxcb             conda-forge/linux-64::libxcb-1.13-h14c3975_1002
  libxml2            conda-forge/linux-64::libxml2-2.9.9-h13577e0_0
  lz4-c              conda-forge/linux-64::lz4-c-1.8.3-he1b5a44_1001
  mapclassify        conda-forge/noarch::mapclassify-2.0.1-py_0
  matplotlib         conda-forge/linux-64::matplotlib-3.1.0-py37_1
  matplotlib-base    conda-forge/linux-64::matplotlib-base-3.1.0-py37hfd891ef_1
  munch              conda-forge/noarch::munch-2.3.2-py_0
  numpy              conda-forge/linux-64::numpy-1.16.4-py37h95a1406_0
  openblas           conda-forge/linux-64::openblas-0.3.5-h9ac9557_1001
  openjpeg           conda-forge/linux-64::openjpeg-2.3.1-h58a6597_0
  pandas             conda-forge/linux-64::pandas-0.24.2-py37hb3f55d8_0
  pcre               conda-forge/linux-64::pcre-8.41-hf484d3e_1003
  pixman             conda-forge/linux-64::pixman-0.38.0-h516909a_1003
  poppler            conda-forge/linux-64::poppler-0.67.0-h2ad7f00_6
  poppler-data       conda-forge/noarch::poppler-data-0.4.9-1
  postgresql         conda-forge/linux-64::postgresql-11.3-hbeee2d4_0
  proj4              conda-forge/linux-64::proj4-6.1.0-he751ad9_2
  pthread-stubs      conda-forge/linux-64::pthread-stubs-0.4-h14c3975_1001
  pyparsing          conda-forge/noarch::pyparsing-2.4.0-py_0
  pyproj             conda-forge/linux-64::pyproj-2.2.1-py37hc44880f_0
  pyqt               conda-forge/linux-64::pyqt-5.9.2-py37hcca6a23_0
  python-dateutil    conda-forge/noarch::python-dateutil-2.8.0-py_0
  pytz               conda-forge/noarch::pytz-2019.1-py_0
  qt                 conda-forge/linux-64::qt-5.9.7-h52cfd70_2
  rtree              conda-forge/linux-64::rtree-0.8.3-py37h666c49c_1002
  scipy              conda-forge/linux-64::scipy-1.3.0-py37h921218d_0
  shapely            conda-forge/linux-64::shapely-1.6.4-py37h06cd6f9_1005
  sip                conda-forge/linux-64::sip-4.19.8-py37hf484d3e_1000
  six                conda-forge/linux-64::six-1.12.0-py37_1000
  tornado            conda-forge/linux-64::tornado-6.0.3-py37h516909a_0
  tzcode             conda-forge/linux-64::tzcode-2019a-h516909a_1002
  xerces-c           conda-forge/linux-64::xerces-c-3.2.2-hac72e42_1001
  xorg-kbproto       conda-forge/linux-64::xorg-kbproto-1.0.7-h14c3975_1002
  xorg-libice        conda-forge/linux-64::xorg-libice-1.0.9-h516909a_1004
  xorg-libsm         conda-forge/linux-64::xorg-libsm-1.2.3-h84519dc_1000
  xorg-libx11        conda-forge/linux-64::xorg-libx11-1.6.7-h14c3975_1000
  xorg-libxau        conda-forge/linux-64::xorg-libxau-1.0.9-h14c3975_0
  xorg-libxdmcp      conda-forge/linux-64::xorg-libxdmcp-1.1.3-h516909a_0
  xorg-libxext       conda-forge/linux-64::xorg-libxext-1.3.4-h516909a_0
  xorg-libxrender    conda-forge/linux-64::xorg-libxrender-0.9.10-h516909a_1002
  xorg-renderproto   conda-forge/linux-64::xorg-renderproto-0.11.1-h14c3975_1002
  xorg-xextproto     conda-forge/linux-64::xorg-xextproto-7.3.0-h14c3975_1002
  xorg-xproto        conda-forge/linux-64::xorg-xproto-7.0.31-h14c3975_1007
  zstd               conda-forge/linux-64::zstd-1.4.0-h3b9ef0a_0
Downloading and Extracting Packages
xorg-renderproto-0.1 | 8 KB      | ##################################### | 100% 
hdf4-4.2.13          | 952 KB    | ##################################### | 100% 
hdf5-1.10.5          | 5.2 MB    | ##################################### | 100% 
xorg-libxrender-0.9. | 31 KB     | ##################################### | 100% 
pcre-8.41            | 249 KB    | ##################################### | 100% 
krb5-1.16.3          | 1.4 MB    | ##################################### | 100% 
zstd-1.4.0           | 928 KB    | ##################################### | 100% 
glib-2.58.3          | 3.3 MB    | ##################################### | 100% 
libcurl-7.64.1       | 584 KB    | ##################################### | 100% 
pandas-0.24.2        | 11.1 MB   | ##################################### | 100% 
xorg-libsm-1.2.3     | 25 KB     | ##################################### | 100% 
munch-2.3.2          | 8 KB      | ##################################### | 100% 
libgfortran-ng-7.3.0 | 1006 KB   | ##################################### | 100% 
cligj-0.5.0          | 8 KB      | ##################################### | 100% 
libdap4-3.19.1       | 1.6 MB    | ##################################### | 100% 
tornado-6.0.3        | 637 KB    | ##################################### | 100% 
kiwisolver-1.1.0     | 86 KB     | ##################################### | 100% 
dbus-1.13.6          | 602 KB    | ##################################### | 100% 
libkml-1.3.0         | 643 KB    | ##################################### | 100% 
sip-4.19.8           | 290 KB    | ##################################### | 100% 
freexl-1.0.5         | 43 KB     | ##################################### | 100% 
qt-5.9.7             | 85.9 MB   | ##################################### | 100% 
pytz-2019.1          | 227 KB    | ##################################### | 100% 
postgresql-11.3      | 4.8 MB    | ##################################### | 100% 
jpeg-9c              | 251 KB    | ##################################### | 100% 
gstreamer-1.14.5     | 4.5 MB    | ##################################### | 100% 
libgdal-2.4.1        | 18.7 MB   | ##################################### | 100% 
freetype-2.10.0      | 885 KB    | ##################################### | 100% 
click-plugins-1.1.1  | 9 KB      | ##################################### | 100% 
xorg-kbproto-1.0.7   | 26 KB     | ##################################### | 100% 
pixman-0.38.0        | 594 KB    | ##################################### | 100% 
libspatialindex-1.9. | 3.1 MB    | ##################################### | 100% 
xorg-libice-1.0.9    | 58 KB     | ##################################### | 100% 
mapclassify-2.0.1    | 24 KB     | ##################################### | 100% 
libcblas-3.8.0       | 6 KB      | ##################################### | 100% 
descartes-1.1.0      | 6 KB      | ##################################### | 100% 
tzcode-2019a         | 429 KB    | ##################################### | 100% 
libtiff-4.0.10       | 587 KB    | ##################################### | 100% 
cycler-0.10.0        | 8 KB      | ##################################### | 100% 
expat-2.2.5          | 191 KB    | ##################################### | 100% 
geopandas-0.5.0      | 881 KB    | ##################################### | 100% 
fiona-1.8.6          | 1.1 MB    | ##################################### | 100% 
matplotlib-base-3.1. | 6.6 MB    | ##################################### | 100% 
curl-7.64.1          | 128 KB    | ##################################### | 100% 
poppler-data-0.4.9   | 3.4 MB    | ##################################### | 100% 
cfitsio-3.430        | 3.1 MB    | ##################################### | 100% 
giflib-5.1.7         | 455 KB    | ##################################### | 100% 
xorg-libxdmcp-1.1.3  | 18 KB     | ##################################### | 100% 
liblapack-3.8.0      | 6 KB      | ##################################### | 100% 
matplotlib-3.1.0     | 6 KB      | ##################################### | 100% 
gst-plugins-base-1.1 | 6.8 MB    | ##################################### | 100% 
libxml2-2.9.9        | 2.0 MB    | ##################################### | 100% 
libnetcdf-4.6.2      | 1.3 MB    | ##################################### | 100% 
click-7.0            | 61 KB     | ##################################### | 100% 
attrs-19.1.0         | 32 KB     | ##################################### | 100% 
geotiff-1.5.1        | 279 KB    | ##################################### | 100% 
icu-58.2             | 22.6 MB   | ##################################### | 100% 
openblas-0.3.5       | 15.8 MB   | ##################################### | 100% 
libssh2-1.8.2        | 257 KB    | ##################################### | 100% 
xorg-libxext-1.3.4   | 51 KB     | ##################################### | 100% 
kealib-1.4.10        | 173 KB    | ##################################### | 100% 
numpy-1.16.4         | 4.3 MB    | ##################################### | 100% 
lz4-c-1.8.3          | 187 KB    | ##################################### | 100% 
pyparsing-2.4.0      | 55 KB     | ##################################### | 100% 
pyproj-2.2.1         | 252 KB    | ##################################### | 100% 
openjpeg-2.3.1       | 470 KB    | ##################################### | 100% 
libspatialite-4.3.0a | 3.1 MB    | ##################################### | 100% 
gettext-0.19.8.1     | 3.6 MB    | ##################################### | 100% 
shapely-1.6.4        | 329 KB    | ##################################### | 100% 
proj4-6.1.0          | 9.4 MB    | ##################################### | 100% 
fontconfig-2.13.1    | 327 KB    | ##################################### | 100% 
xorg-libxau-1.0.9    | 13 KB     | ##################################### | 100% 
gdal-2.4.1           | 1.4 MB    | ##################################### | 100% 
pyqt-5.9.2           | 5.7 MB    | ##################################### | 100% 
libedit-3.1.20170329 | 172 KB    | ##################################### | 100% 
scipy-1.3.0          | 18.8 MB   | ##################################### | 100% 
libblas-3.8.0        | 6 KB      | ##################################### | 100% 
libiconv-1.15        | 2.0 MB    | ##################################### | 100% 
poppler-0.67.0       | 8.8 MB    | ##################################### | 100% 
python-dateutil-2.8. | 219 KB    | ##################################### | 100% 
xorg-libx11-1.6.7    | 940 KB    | ##################################### | 100% 
rtree-0.8.3          | 48 KB     | ##################################### | 100% 
boost-cpp-1.70.0     | 21.0 MB   | ##################################### | 100% 
pthread-stubs-0.4    | 5 KB      | ##################################### | 100% 
libpng-1.6.37        | 343 KB    | ##################################### | 100% 
xerces-c-3.2.2       | 3.3 MB    | ##################################### | 100% 
geos-3.7.1           | 1.6 MB    | ##################################### | 100% 
six-1.12.0           | 22 KB     | ##################################### | 100% 
libpq-11.3           | 2.5 MB    | ##################################### | 100% 
libxcb-1.13          | 396 KB    | ##################################### | 100% 
libuuid-2.32.1       | 26 KB     | ##################################### | 100% 
xorg-xextproto-7.3.0 | 27 KB     | ##################################### | 100% 
json-c-0.13.1        | 71 KB     | ##################################### | 100% 
cairo-1.16.0         | 1.5 MB    | ##################################### | 100% 
xorg-xproto-7.0.31   | 72 KB     | ##################################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
58.44s$ conda install --yes --file requirements.txt;
Collecting package metadata (current_repodata.json): done
Solving environment: failed
Collecting package metadata (repodata.json): done
Solving environment: failed
UnsatisfiableError: The following specifications were found to be incompatible with each other:
  - conda-forge/linux-64::boost-cpp==1.70.0=ha2d47e9_0 -> libboost[version='<0']
  - conda-forge/linux-64::libkml==1.3.0=h4fcabce_1010 -> boost-cpp[version='>=1.70.0,<1.70.1.0a0'] -> libboost[version='<0']
  - conda-forge/linux-64::python==3.7.3=h5b0a415_0
  - osmnx -> geopy
  - urbanaccess -> geopy[version='>=1.11.0']
Note that strict channel priority may have removed packages required for satisfiability.
The command "conda install --yes --file requirements.txt;" failed and exited with 1 during .
Your build has been stopped.
```

renanxcortes commented:
So, I think that after merging https://github.com/pysal/segregation/pull/78, perhaps this will not be a problem anymore since `osmnx` and `urbanaccess` will be dropped from requirements.txt.
